addons/website_sale/models/product_product.py

```diff
@@ -103,4 +103,6 @@ def _get_contextual_price_tax_selection(self):
         company_taxes = self.taxes_id.filtered(lambda tax: tax.company_id == self.env.company)
         if line_tax_type == "tax_included" and company_taxes:
             price = company_taxes.compute_all(price, product=self, partner=self.env['res.partner'])['total_included']
+        if line_tax_type == "tax_excluded" and any(company_taxe.price_include for company_taxe in company_taxes):
+            price = company_taxes.compute_all(price, product=self, partner=self.env['res.partner'])['total_excluded']
```

Comment:
Why don't we always compute the taxes (if there are taxes) and take either `total_included` or `total_excluded` according to the configuration then ?

Also, after more thoughts, I find it strange that here we use the ICP to know whether we want tax-inc or tax-excl when in `_price_with_tax_computed` we check the user groups.  Not your problem at all, but there's probably something I should clean/clarify (and eventually use `_price_with_tax_computed` here as well)

Reply:
```suggestion
        fpos_id = self.env['website'].sudo()._get_current_fiscal_position_id(partner)
        fiscal_position = self.env['account.fiscal.position'].sudo().browse(fpos_id)
        product_taxes = self.sudo().taxes_id.filtered(lambda x: x.company_id == company_id)
        taxes = fiscal_position.map_tax(product_taxes)
        if product_taxes:
            price = self.env['account.tax']._fix_tax_included_price_company(
                price, product_taxes, taxes, self.env.company.id,
            )
            tax_display = "total_included" if line_tax_type == "tax_included" else "total_excluded"
            price = company_taxes.compute_all(price, product=self, partner=self.env['res.partner'])[tax_display]
```

Looking at it more deeply, I believe some other points are wrong and need to be fixed as well:
1) fiscal position
2) _fix_tax_included_price_company logic